pkg/cmd/experimental/syncgroups/schema1/newsyncer.go

```diff
@@ -0,0 +1,66 @@
+package schema1
+
+import (
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups"
+	syncutil "github.com/openshift/origin/pkg/cmd/experimental/syncgroups/util"
+)
+
+// NewGroupSyncer creates a GroupSyncer for a schema-specific config written for an LDAP schema with
+// first-class group entries that hold membership lists as attributes
+func NewGroupSyncer(clientConfig ldaputil.ClientConfig,
+	ldapUserAttributeDefiner ldaputil.LDAPUserAttributeDefiner,
+	userDefinedGroupNameMapping bool,
+	ldapGroupUIDToOpenShiftNameMapping map[string]string,
+	specificGroupQueryTemplate syncutil.LDAPSpecificQueryTemplate,
+	specificUserQueryTemplate syncutil.LDAPSpecificQueryFromAttributeTemplate,
+	groupNameAttribute []string,
+	scope syncgroups.GroupSyncScope,
+	source syncroups.GroupSyncSource,
+	client client.GroupInterface,
+	whitelistContents []string) syncgroups.GroupSyncer {
+	// the schema-specific ldapInterface is built from the config
+	ldapInterface := NewLDAPInterface(clientConfig,
+		ldapUserAttributeDefiner,
+		specificGroupQueryTemplate,
+		specificUserQueryTemplate)
+
+	syncer := syncgroups.LDAPGroupSyncer{
+		Host:                 clientConfig.Host,
+		GroupClient:          client,
+		UserNameMapper:       syncgroups.NewUserNameMapper(ldapUserAttributeDefiner),
+		GroupMemberExtractor: &ldapInterface,
+	}
+
+	// In order to build the GroupNameMapper, we need to know if the user defined a hard mapping
+	// or one based on LDAP group entry attributes
+	if ldapSyncConfig.CommonConfig.UserDefinedGroupNameMapping {
```

Comment:
also this refactor into a method ends up being just as long in terms of lines of code. make your choice, I'm fairly ambivalent